File: frontend/mobile_client/lib/features/jobs/domain/entities/job.dart
class Job {
  final int? id;
  final String title;
  final String description;
  final String company;
  final String location;
  final String type;
  final int postedBy;
  final String posterName;
  final String? createdAt;
  final String? updatedAt;

  const Job({
    this.id,
    required this.title,
    required this.description,
    required this.company,
    required this.location,
    required this.type,
    required this.postedBy,
    required this.posterName,
    this.createdAt,
    this.updatedAt,
  });
}

File: backend/job-service/src/main/java/com/decp/job/model/Job.java
package com.decp.job.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "jobs")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Job {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String company;
    private String location;
    private String type; // e.g., Internship, Full-time
    
    private Long postedBy; // User ID of the alumni/admin who posted
    private String posterName;
    
    private String status; // OPEN or CLOSED
    private Integer applicationCount; // Number of applications

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
        if (status == null) {
            status = "OPEN";
        }
        if (applicationCount == null) {
            applicationCount = 0;
        }
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
